### ISSUE TITLE
Supported API version?

### ISSUE THREAD

**Issue by python273** [dcoles/flipperzero-rs#15]
Do I understand correctly 7.5 is not supported?

**Comment by dcoles** [comment]
That's correct. I have a WIP branch ([`sdk7.x`](https://github.com/dcoles/flipperzero-rs/tree/sdk7.x)), which brings everything up to date; but the SDK version check fails on newer firmware and I'm still trying to debug that.

**Comment by python273** [comment]
Cool. I manually updated `API_VERSION` to match header from another compiled app (should be `7.51` from [unleashed](https://github.com/DarkFlippers/unleashed-firmware)), it passes version check, but the app stops immediately
```diff
diff --git a/crates/sys/src/bindings.rs b/crates/sys/src/bindings.rs
index a38fb8b..e4c3087 100644
--- a/crates/sys/src/bindings.rs
+++ b/crates/sys/src/bindings.rs
@@ -110,7 +110,7 @@ impl<T> ::core::fmt::Debug for __IncompleteArrayField<T> {
         fmt.write_str("__IncompleteArrayField")
     }
 }
-pub const API_VERSION: u32 = 458752;
+pub const API_VERSION: u32 = 3342343;
 #[repr(C)]
 #[derive(Debug, Copy, Clone)]
 pub struct Bt {
```

**Comment by python273** [comment]
I guess major and minor should be in the reverse order
https://github.com/dcoles/flipperzero-rs/blob/57df18579b92b9340103a4958676291657a07c8d/tools/src/bin/generate-bindings.rs#L58

**Comment by python273** [comment]
> app stops immediately

oh, that's what it supposed to do 😅 figured out how to see logs via serial, prints `Hello, Rust!`, so seems to be working

**Comment by dcoles** [comment]
> I guess major and minor should be in the reverse order

Ah, thank you! I forgot that endianess causes all sorts of havoc when dealing with type punning.
If that fixes everything, I should be able to tag a new version shortly.

**Comment by dcoles** [comment]
Released flipperzero-rs v0.4.0 with support for SDK 7.5 (see [crates.io](https://crates.io/crates/flipperzero/0.4.0)).

Let me know if you have any issue.
